File: elixir/get.exs
defmodule Get do
  def run(url) do
    :ok = :inets.start()

    :httpc.request(url)
    |> parse()
  end

  defp parse({:ok, {{_, 200, _}, _headers, body}}) do
    IO.puts(body)
  end

  defp parse({:ok, {_, status, _}, _headers, body}) do
    IO.puts("Error! Status: #{status}")
    IO.puts("Body: #{body}")
  end

  defp parse({:error, err}),
    do: IO.puts("Request error: #{err}")
end

Get.run("http://httpbin.org")

File: js/nodejs/get.mjs
import http from "http";

const URL = "your_endpoint";

http.get(URL, (res) => {
  const { statusCode } = res;
  const contentType = res.headers["content-type"];

  if (statusCode !== 200)
    throw new Error(`Request Failed.\n Status: ${statusCode}`);

  res.setEncoding("utf-8");
  
  let rawData = "";
  res.on("data", (chunk) => { rawData += chunk });

  res.on("end", () => console.log(rawData));

}).on("error", (err) => console.error(`An error: ${err.message}`));
